src/documents/ResourceLoader.md

```diff
@@ -4,114 +4,261 @@ category: Infusion
+[initial transaction](ModelRelay.md#the-initial-transaction). Other kinds of "fast path" may be made available by
+other core framework grades, for example the upcoming "new renderer".
+
+Any resource not loaded via the "fast path" will be loaded via the "slow path" and will have its fetch triggered by
+the component's ``onCreate`` event.
+
+When all resources, via all routes, configured for the component, have been loaded, the resource loader fires
+an ```onResourcesLoaded``` event with an argument of the populated ``resources`` structure. These will each have fields
+filled in corresponding to the loaded resource, most importantly the field ``parsed`` holding a parsed (e.g. as JSON)
+representation of the resource, and also the original fetched resource text within the field ``resourceText``.
+This populated structure can also be retrieved directly on the resource loader instance via the path ```resources```.
+
+Note that even if all of the component's resources have already been loaded via the "fast path", ``onResourcesLoaded`` will
+only fire once the component's ``onCreate`` event has concluded. If any resource loading fails, the event ``onResourceError``
+will be fired for the failing resource, and ``onResourcesLoaded`` will not fire.
+
+The underlying implementation of ``fluid.resourceLoader`` is the low-level ``fluid.resourceFetcher`` mini-component.
+In the browser, for URL resources, this operates a native [XMLHttpRequest](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest),
+whereas in node.js this operates the native node [HTTP](https://nodejs.org/api/http.html) or [HTTPS request API](https://nodejs.org/api/https.html).
+For all transports, there is the possibility to pass arbitrary options through the ResourceLoader configuration
+structure to take full control of the loading process, whilst there is a central core of portable options supported
+in all environments.
+
+### ``fluid.resourceLoader`` options
+
+We name the subset of component options particularly consumed by ``fluid.resourceLoader`` as ``resourceLoaderOptions``.
+``resourceLoaderOptions`` contains two top-level keys, ``resourceOptions``, and ``resources``, which are both optional.
+
+Schematically, ``fluid.resourceLoader`` configuration looks like
+
+```snippet
+fluid.resourceLoader({
+    resourceOptions: <resourceOptions>
+    resources: {
+        resourceKey1: <resourceSpec>
+        resourceKey2: <resourceSpec>
+    ....
+    }
+})
+```
+
+In short form, a ``resourceSpec`` may simply consist of a String, in which case it is interpreted as a URL. Otherwise,
+it is an Object with keys described in the [section on resourceSpecs](#resourcespec-structure).
+
+### `resourceOptions` structure
 
 <table>
+    <tr>
+        <th colspan="3">Members within the <code>resourceOptions</code> structure at top level in the options
+        <code>fluid.resourceLoader</code> component</th>
+    </tr>
```

Comment:
I don't know if this should be part of the table. It spans all of the columns, and I'm wondering if it should could be before the table, or somehow associated as a description.

Reply:
This is the same approach we've used in tables elsewhere, e.g. ChangeApplierAPI.md - if we think a change is helpful we should file a dedicated issue and review globally

Reply:
Filed as #218